packages/common/src/types/rejectReason.ts

```diff
@@ -0,0 +1,200 @@
+    | RejectReasonTag.ModuleNotWF
+    | RejectReasonTag.RuntimeFailure
+    | RejectReasonTag.SerializationFailure
+    | RejectReasonTag.OutOfEnergy
+    | RejectReasonTag.InvalidProof
+    | RejectReasonTag.InsufficientBalanceForBakerStake
+    | RejectReasonTag.StakeUnderMinimumThresholdForBaking
+    | RejectReasonTag.BakerInCooldown
+    | RejectReasonTag.NonExistentCredentialID
+    | RejectReasonTag.KeyIndexAlreadyInUse
+    | RejectReasonTag.InvalidAccountThreshold
+    | RejectReasonTag.InvalidCredentialKeySignThreshold
+    | RejectReasonTag.InvalidEncryptedAmountTransferProof
+    | RejectReasonTag.InvalidTransferToPublicProof
+    | RejectReasonTag.InvalidIndexOnEncryptedTransfer
+    | RejectReasonTag.ZeroScheduledAmount
+    | RejectReasonTag.NonIncreasingSchedule
+    | RejectReasonTag.FirstScheduledReleaseExpired
+    | RejectReasonTag.InvalidCredentials
+    | RejectReasonTag.RemoveFirstCredential
+    | RejectReasonTag.CredentialHolderDidNotSign
+    | RejectReasonTag.NotAllowedMultipleCredentials
+    | RejectReasonTag.NotAllowedToReceiveEncrypted
+    | RejectReasonTag.NotAllowedToHandleEncrypted
+    | RejectReasonTag.MissingBakerAddParameters
+    | RejectReasonTag.FinalizationRewardCommissionNotInRange
+    | RejectReasonTag.BakingRewardCommissionNotInRange
+    | RejectReasonTag.TransactionFeeCommissionNotInRange
+    | RejectReasonTag.AlreadyADelegator
+    | RejectReasonTag.InsufficientBalanceForDelegationStake
+    | RejectReasonTag.MissingDelegationAddParameters
+    | RejectReasonTag.InsufficientDelegationStake
+    | RejectReasonTag.DelegatorInCooldown
+    | RejectReasonTag.StakeOverMaximumThresholdForPool
+    | RejectReasonTag.PoolWouldBecomeOverDelegated
+    | RejectReasonTag.PoolClosed;
+
+export type ModuleRefRejectReasonTag =
+    | RejectReasonTag.ModuleHashAlreadyExists
+    | RejectReasonTag.InvalidModuleReference;
+
+export type AccountAddressRejectReasonTag =
+    | RejectReasonTag.InvalidAccountReference
+    | RejectReasonTag.NotADelegator
+    | RejectReasonTag.NonExistentRewardAccount
+    | RejectReasonTag.NotABaker
+    | RejectReasonTag.ScheduledSelfTransfer
+    | RejectReasonTag.EncryptedAmountSelfTransfer;
+
+export type StringRejectReasonTag =
+    | ModuleRefRejectReasonTag
+    | AccountAddressRejectReasonTag
+    | RejectReasonTag.NonExistentCredentialID
+    | RejectReasonTag.DuplicateAggregationKey;
+
+export interface StringRejectReason {
+    tag: StringRejectReasonTag;
+    contents: HexString | Base58String;
+}
+export type NumberRejectReasonTag =
+    | RejectReasonTag.AlreadyABaker
+    | RejectReasonTag.DelegationTargetNotABaker;
+
+export interface NumberRejectReason {
+    tag: NumberRejectReasonTag;
+    contents: number;
+}
+
+export interface SimpleRejectReason {
+    tag: SimpleRejectReasonTag;
+}
+
+export interface InvalidReceiveMethod {
+    tag: RejectReasonTag.InvalidReceiveMethod;
+    contents: [HexString, string]; // [moduleRef, receiveName]
```

Comment:
Is this "contents" so that you support the JSON format for V1? Because it is not the best API for users. It would be much better to have named fields.

Reply:
Yeah, the reject reasons are using the same format as v1.

Reply:
But, we'll change it.